2007-12-01  Jim Meyering  <meyering@redhat.com>

	Fix a bug that inhibited much of the utf8-optimization in regcomp.c.
	* lib/regcomp.c (optimize_utf8): Fix a typo, s/idx/ctx_type/,
	that would inhibit utf8-optimization of a regexp containing line-
	or buffer-anchors, e.g., `^', `$'.

---
 posix/regcomp.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: glibc-2.7/posix/regcomp.c
===================================================================
--- glibc-2.7.orig/posix/regcomp.c
+++ glibc-2.7/posix/regcomp.c
@@ -1030,7 +1030,7 @@ optimize_utf8 (re_dfa_t *dfa)
 	  mb_chars = 1;
 	break;
       case ANCHOR:
-	switch (dfa->nodes[node].opr.idx)
+	switch (dfa->nodes[node].opr.ctx_type)
 	  {
 	  case LINE_FIRST:
 	  case LINE_LAST:
